rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -161,4 +174,28 @@ public static boolean tableExists(SQLiteDatabase db, String tableName) {
     public static boolean isEmptyCursor(Cursor cursor) {
         return cursor == null || cursor.getCount() == 0;
     }
+
+    @Nullable
+    public static String getParentLocationId() {
+        AllSharedPreferences allSharedPreferences = RDTApplication.getInstance().getContext().allSharedPreferences();
+        String defaultLocationUuid = allSharedPreferences.fetchDefaultLocalityId(allSharedPreferences.fetchRegisteredANM());
+        LinkedHashMap<String, TreeNode<String, Location>> locationMap = LocationHelper.getInstance().map();
+        return getParentLocationId(defaultLocationUuid, locationMap);
+    }
+
+    @Nullable
+    public static String getParentLocationId(String childLocationId, LinkedHashMap<String, TreeNode<String, Location>> map) {
+        Map.Entry<String, TreeNode<String, Location>> entry = map.entrySet().iterator().next();
+        if (entry != null) {
+            if (entry.getKey().equals(childLocationId)) {
+                return entry.getValue().getParent();
+            }
+            else {
+                return getParentLocationId(childLocationId, entry.getValue().getChildren());
+            }
+        }
+        else {
+            return null;
+        }
+    }
```

Comment:
Can this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L588  and this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L596 be used to get the parent location id?